e2e/rescheduling/rescheduling_test.go

```diff
@@ -0,0 +1,543 @@
+		must.Sprintf("expected only 1 failed deployment, got:\n%s", out))
+	must.Eq(t, 2, statuses["successful"],
+		must.Sprintf("expected 2 successful deployments, got:\n%s", out))
+}
+
+// TestRescheduling_ProgressDeadline verifies the progress deadline is reset with
+// each healthy allocation, and that a rescheduled allocation does not.
+func TestRescheduling_ProgressDeadline(t *testing.T) {
+
+	jobID := "test-reschedule-deadline-" + uuid.Generate()[0:8]
+	must.NoError(t, e2eutil.Register(jobID, "./input/rescheduling_progressdeadline.nomad"))
+
+	t.Cleanup(func() {
+		e2eutil.StopJob(jobID, "-purge", "-detach")
+		_, err := e2eutil.Command("nomad", "system", "gc")
+		test.NoError(t, err)
+	})
+
+	expected := []string{"running"}
+	must.NoError(t,
+		e2eutil.WaitForAllocStatusExpected(jobID, ns, expected),
+		must.Sprint("should have a running allocation"),
+	)
+
+	deploymentID, err := e2eutil.LastDeploymentID(jobID, ns)
+	must.NoError(t, err, must.Sprint("couldn't look up deployment"))
+
+	oldDeadline, err := getProgressDeadline(deploymentID)
+	must.NoError(t, err, must.Sprint("could not get progress deadline"))
+	time.Sleep(time.Second * 20)
+
+	newDeadline, err := getProgressDeadline(deploymentID)
+	must.NoError(t, err, must.Sprint("could not get new progress deadline"))
+	must.NotEq(t, oldDeadline, newDeadline,
+		must.Sprint("progress deadline should have been updated"))
+
+	must.NoError(t, e2eutil.WaitForLastDeploymentStatus(jobID, ns, "successful", nil),
+		must.Sprint("deployment should be successful"))
+}
+
+// TestRescheduling_ProgressDeadlineFail verifies the progress deadline is reset with
+// each healthy allocation, and that a rescheduled allocation does not.
+func TestRescheduling_ProgressDeadlineFail(t *testing.T) {
+
+	jobID := "test-reschedule-deadline-fail" + uuid.Generate()[0:8]
+	must.NoError(t, e2eutil.Register(jobID, "./input/rescheduling_progressdeadline_fail.nomad"))
+
+	t.Cleanup(func() {
+		e2eutil.StopJob(jobID, "-purge", "-detach")
+		_, err := e2eutil.Command("nomad", "system", "gc")
+		test.NoError(t, err)
+	})
+
+	testutil.WaitForResult(func() (bool, error) {
+		_, err := e2eutil.LastDeploymentID(jobID, ns)
+		return err == nil, err
+	}, func(err error) {
+		must.NoError(t, err, must.Sprint("deployment wasn't created yet"))
+	})
+
+	deploymentID, err := e2eutil.LastDeploymentID(jobID, ns)
+	must.NoError(t, err, must.Sprint("couldn't look up deployment"))
+
+	oldDeadline, err := getProgressDeadline(deploymentID)
+	must.NoError(t, err, must.Sprint("could not get progress deadline"))
+	time.Sleep(time.Second * 20)
+
+	must.NoError(t, e2eutil.WaitForLastDeploymentStatus(jobID, ns, "failed", nil),
+		must.Sprint("deployment should be failed"))
+
+	must.NoError(t,
+		e2eutil.WaitForAllocStatusComparison(
+			func() ([]string, error) { return e2eutil.AllocStatuses(jobID, ns) },
+			func(got []string) bool {
+				for _, status := range got {
+					if status != "failed" {
+						return false
+					}
+				}
+				return true
```

Comment:
returning `false` within the for loop, then `true` after it like this would fix (the first part of) my comment on the `TestRescheduling_Success` test up above (swapping `!= "failed"` for `!= "running"`).  this one does appropriately confirm that _all_ allocs are "failed"